preinstall.sh

```diff
@@ -65,6 +119,121 @@ if [ "$(id -u)" -ne "0" ]; then
+            apt-get install -y automake build-essential curl zlib1g-dev libopenjpeg-dev libpng12-dev libjpeg-dev libtiff5-dev libgdk-pixbuf2.0-dev libxml2-dev libsqlite3-dev libcairo2-dev libglib2.0-dev sqlite3 libsqlite3-dev
+            install_libopenslide_from_source
+            ;;
+          precise|wheezy|maya)
+            # Debian 7, Ubuntu 12.04, Mint 13
+            echo "Installing libopenslide dependencies via apt-get"
+            apt-get install -y automake build-essential curl zlib1g-dev libopenjpeg-dev libpng12-dev libjpeg-dev libtiff5-dev libgdk-pixbuf2.0-dev libxml2-dev libsqlite3-dev libcairo2-dev libglib2.0-dev sqlite3 libsqlite3-dev
+            install_libopenslide_from_source
+            ;;
+          *)
+            # Unsupported Debian-based OS
+            sorry "openslide" "Debian-based $DISTRO"
+            ;;
+        esac
+      elif [ -f /etc/redhat-release ]; then
+        # Red Hat Linux
+        RELEASE=$(cat /etc/redhat-release)
+        echo "Detected Red Hat Linux '$RELEASE'"
+        case $RELEASE in
+          "Red Hat Enterprise Linux release 7."*|"CentOS Linux release 7."*|"Scientific Linux release 7."*)
+            # RHEL/CentOS 7
+            echo "Installing libopenslide dependencies via yum"
+            yum groupinstall -y "Development Tools"
+            yum install -y tar curl libpng-devel libjpeg-devel libxml2-devel zlib-devel openjpeg-devel libtiff-devel gdk-pixbuf2-devel sqlite-devel cairo-devel glib2-devel
+            install_libopenslide_from_source "--prefix=/usr"
+            ;;
+          "Red Hat Enterprise Linux release 6."*|"CentOS release 6."*|"Scientific Linux release 6."*)
+            # RHEL/CentOS 6
+            echo "Installing libopenslide dependencies via yum"
+            yum groupinstall -y "Development Tools"
+            yum install -y tar curl libpng-devel libjpeg-devel libxml2-devel zlib-devel openjpeg-devel libtiff-devel gdk-pixbuf2-devel sqlite-devel cairo-devel glib2-devel 
+            install_libopenslide_from_source "--prefix=/usr"
+            ;;
+          "Fedora release 21 "*|"Fedora release 22 "*)
+            # Fedora 21, 22
+            echo "Installing libopenslide via yum"
+            yum install -y openslide-devel
+            ;;
+          *)
+            # Unsupported RHEL-based OS
+            sorry "openslide" "$RELEASE"
+            ;;
+        esac
+      elif [ -f /etc/system-release ]; then
+        # Probably Amazon Linux
+        RELEASE=$(cat /etc/system-release)
+        case $RELEASE in
+          "Amazon Linux AMI release 2014.09")
+            # Amazon Linux
+            echo "Detected '$RELEASE'"
+            echo "Installing libopenslide dependencies via yum"
+            yum groupinstall -y "Development Tools"
+            yum install -y tar curl libpng-devel libjpeg-devel libxml2-devel zlib-devel openjpeg-devel libtiff-devel gdk-pixbuf2-devel sqlite-devel cairo-devel glib2-devel 
```

Comment:
I've just discovered AWS doesn't provide `gdk-pixbuf2-devel` in their repo, so this step generates a warning and then the compilation fails. Probably best to remove the whole "Amazon Linux" section for openslide.